drag_and_drop_v2/public/js/drag_and_drop_edit.js

```diff
@@ -183,12 +183,22 @@ function DragAndDropEditBlock(runtime, element) {
                         remove: function(e) {
                             var $el = $(e.currentTarget).closest('.zone-row'),
                                 classes = $el.attr('class'),
-                                id = classes.slice(classes.indexOf('zone-row') + 9);
+                                id = classes.slice(classes.indexOf('zone-row') + 9),
+                                // The id is of the form "zone-<index>".  We extract the index by
+                                // slicing off the first five letters and "casting" to integer.
+                                index = +id.slice(5),
```

Comment:
@smarnach this will break if any other class added after 'zone-<index>' (e.g. `zone-row zone-2 zone-disabled`). +1 for @bradenmacdonald 's suggestion below